docs/apis/plugintypes/format/index.mdx

```diff
@@ -0,0 +1,613 @@
+
+import LibExample from '!!raw-loader!./_examples/lib.php';
+
+<Lib
+    plugintype="format"
+    pluginname="pluginname"
+    example={LibExample}
+    // Not legacy for format.
+    legacy={false}
+    required
+>
+The main library of the format. It should contain a class format_pluginname extending core_courseformat\base, this class is known as the format base class. Also, it may contain callbacks for other core and contributed APIs if necessary.
+</Lib>
+
+The format base class is the most important part of the course format as it defines how the format interacts with both frontend and backend. Depending on the methods a format plugin overrides the course will behave in different ways.
+
+### lang/en/format_pluginname.php
+
+import langExample from '!!raw-loader!./_examples/format_lang.php';
+
+<Lang
+    plugintype="format"
+    pluginname="pluginname"
+    example={langExample}
+/>
+
+### classes/output/renderer.php
+
+import Renderer from '!!raw-loader!./_examples/renderer.php';
+
+<ComponentFileSummary
+    required={true}
+    filepath="/format.php"
+    summary="Format file course rendering"
+    plugintype="format"
+    pluginname="pluginname"
+    example={Renderer}
+    required
+    description="Unlike the rest of the Moodle plugins, all course format plugins must provide a renderer."
+/>
+
+This file should:
+- Use the namespace format_pluginname\output (all files in the classes folder should be namespaced)
+- The class should be called renderer (all files in the classes folder match the name of the file)
+- Should extend core_courseformat\output\section_renderer. See Output renderers for more information.
+
+Renderer methods are powerful enough to override all the rendering logic of the course. However, by default, the way to render a course should be based on output classes and templates (see the [output structure section](#format-output-classes-and-templates) for more information).
+
+Since the course format is all about the display it is very important to separate HTML and PHP. All HTML code should be in your format_pluginname_renderer class in renderer.php. Ideally, format.php will only call one function from the renderer. Use of renderer is required if you want to output content header and footer.
+
+### version.php
+
+<VersionPHP
+    plugintype="format"
+/>
+
+### classes/output/courseformat/*
+
+Since Moodle 4.0 the course rendering is based on output classes and templates. Course format plugins can override specific output classes to provide alternative data to the templates. All output classes inside output/courseformat folder will override the default ones automátically. See the output structure section below for more information
+
+### templates/* and templates/local/*
+
+Since Moodle 4.0 the course rendering is based on output classes and templates. This folder will contain the specific mustache templates of your plugin. See the [override mustache blocks section](#override-mustache-blocks) for more information.
```

Comment:
```suggestion
Since Moodle 4.0 the course rendering is based on output classes and templates. This folder will contain the specific mustache templates of your plugin. See the [override mustache blocks section](#override-mustache-blocks) section for more information.
```